exchanges/orderbook/validate.go

```diff
@@ -0,0 +1,96 @@
+package orderbook
+
+import (
+	"fmt"
+
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+// checker defines specific functionality to determine ascending/descending
+// validation
+type checker func(current, previous Level) error
+
+// asc specifically defines ascending price check
+var asc = func(current, previous Level) error {
+	if current.Price < previous.Price {
+		return errPriceOutOfOrder
+	}
+	return nil
+}
+
+// dsc specifically defines descending price check
+var dsc = func(current, previous Level) error {
+	if current.Price > previous.Price {
+		return errPriceOutOfOrder
+	}
+	return nil
+}
+
+// Validate ensures that the orderbook items are correctly sorted and all fields are valid
+// Bids should always go from a high price to a low price and Asks should always go from a low price to a higher price
+func (b *Book) Validate() error {
+	if !b.VerifyOrderbook {
+		return nil
+	}
+	return validate(b)
+}
+
+func validate(b *Book) error {
+	// Checking for both ask and bid lengths being zero has been removed and
+	// a warning has been put in place for some exchanges that return zero
+	// level books. In the event that there is a massive liquidity change where
+	// a book dries up, this will still update so we do not traverse potential
+	// incorrect old data.
```

Comment:
🚧 Can rephrase this to a permanent perspective, please?
I kind of understand what it's saying, but even looking at the introductory commit eb0571cc9 I'm not sure what was removed exactly.

Now that I'm looking at this more, I think the second comment no longer applies at all.
We're talking about "this will still update", but we're not going to update anything that I can see.

So ... saying that we're just warning and it's not an error is self-evident.
The biggest thing I think I'd want to know is why options are exempt.

Something like:
```suggestion
	// Some exchanges may return empty sides, but it's not an error
	// Options have empty sides too frequently for this warning to be useful
```

🚧 I'm not sure if this comment relates to the above or the next lines of code
	// In the event that there is a massive liquidity change where a book dries up, this will still update so we do not traverse potential incorrect old data.
```